diff --git a/pymavlink/generator/mavparse.py b/pymavlink/generator/mavparse.py
index 23ce40e..ac7a712 100644
--- a/pymavlink/generator/mavparse.py
+++ b/pymavlink/generator/mavparse.py
@@ -6,7 +6,7 @@ Copyright Andrew Tridgell 2011
 Released under GNU GPL version 3 or later
 '''
 
-import xml.parsers.expat, os, time, sys, operator, mavutil
+import xml.parsers.expat, os, errno, time, sys, operator, mavutil
 
 PROTOCOL_0_9 = "0.9"
 PROTOCOL_1_0 = "1.0"
@@ -359,16 +359,12 @@ def total_msgs(xml):
     return count
 
 def mkdir_p(dir):
-    '''like mkdir -p'''
-    if not dir:
-        return
-    if dir.endswith("/"):
-        mkdir_p(dir[:-1])
-        return
-    if os.path.isdir(dir):
-        return
-    mkdir_p(os.path.dirname(dir))
-    os.mkdir(dir)
+    try:
+        os.makedirs(dir)
+    except OSError as exc:
+        if exc.errno == errno.EEXIST:
+            pass
+        else: raise
 
 # check version consistent
 # add test.xml
